### ISSUE TITLE
webkitSupportsPresentationMode throwing error

### ISSUE THREAD

**Issue by Curetix** [vidstack/player#989]
### Current Behavior:

The player (sometimes) throws the following error:

![image](https://github.com/vidstack/player/assets/15160542/f0de00ce-b9e2-4001-981c-0e1e6a53315c)

Here's the line of code:

![image](https://github.com/vidstack/player/assets/15160542/a3385af3-33b5-4736-b099-f33da8715539)

The player still works fine after the error is thrown. Happening both in dev and production mode.

### Environment:

- Vidstack: 1.5.1
- Framework: React 18
- Meta Framework: Next.js 14
- Node: 20
- OS: Windows 11
- Browser: Brave

**Comment by mihar-22** [comment]
Did this just start in `1.5.1`? Does this happen when doing anything specific like navigating to a new route? Any other info you can provide?

**Comment by Curetix** [comment]
> Did this just start in `1.5.1`? Does this happen when doing anything specific like navigating to a new route? Any other info you can provide?

The error started popping up with 1.4.8 or 1.4.9, not 100% sure. From what I can tell it's happening when navigating to a route for the first time, but only when the player uses a live source.

**Comment by cbravo** [comment]
I am seeing the same thing using version 1.5.7 when I load a route from scratch. I am using Nextjs 14.0.3

**Comment by cbravo** [comment]
It seems like all this did was move the problem to a different place.
Now I am getting 
`TypeError: Cannot read properties of null (reading 'disablePictureInPicture')`
 This may have been just a symptom of a deeper problem? Is it normal that this ref be undefined? I definitely have the media provider component so I am not sure what is going on here.
![image](https://github.com/vidstack/player/assets/5489389/636680cc-f068-4cac-ae4a-f5aab7f7094e)

**Comment by mihar-22** [comment]
It shouldn't be `null` on mount. I'll look into it after a few other things.

**Comment by cbravo** [comment]
The problem seems to be related to re-rendering some parent component right away using a useEffect. I have a context I am using that has a useEffect that causes a second render right away and its on this second render that the ref is undefined Or at least this somehow fits into the issue.

**Comment by cbravo** [comment]
Okay I believe the problem is happening when you unmount the video on very first useEffect. I realized that the useEffect I referenced above is moving the position of the video in the dom and so I think vidstack is still doing something behind the scenes on the unmounted video and thats why the ref is null. Should there be some kind of check to see if the video is still mounted or something along those lines?
